### DISCUSSION TITLE
Add support for enter key/submit in Input component

### DISCUSSION THREAD

**Discussion by pranshuchittora** [twilio-labs/paste#2831] [Ideas] [answered]
Currently the `Input` component doesn't provides an interface/prop to call a function when a user presses enter key or submit in mobile. 

Possible solution 
- Add a prop named onPressEnter like antd or onSubmit or something related and relevant.

https://ant.design/components/input/

<img width="1943" alt="Screenshot 2022-11-15 at 23 41 51" src="https://user-images.githubusercontent.com/32242596/201995100-61c677f9-0fa9-4184-951f-b31b4532a84d.png">

Good Day

**Comment by SiTaggart** [comment, marked as answer]
Hi @pranshuchittora 

We rely heavily on native web platform conventions. As such, the things you are trying to achieve can be done as if you were just using HTML on its own.

To respond to an enter key press inside a form control, you should use some of the `keyPress` or `keyDown` events that are support by the HTML Input element.

To respond to a form submission, you should use a form element to wrap your form, and respond to the `onSubmit` form event.

I've created a Codesandbox to demonstrate how this works 

https://codesandbox.io/s/form-events-and-submission-example-e33kbf
